daemon/c_service.c

```diff
@@ -511,18 +527,58 @@ c_service_start_pre_exec(c_service_t *service)
  * Helper function that generates and sends a protobuf message to the Trustme Service.
  */
 static int
-c_service_send_message_proto(c_service_t *service, unsigned int code)
+c_service_send_message_proto(c_service_t *service, unsigned int code, const char *msg)
 {
 	ASSERT(service);
 
 	CmldToServiceMessage message_proto = CMLD_TO_SERVICE_MESSAGE__INIT;
 	message_proto.code = code;
 
+	if (msg) {
+		message_proto.msg = mem_strdup(msg);
```

Comment:
could it be that there is an allready allocated memory which will later,
be free'd externaly? Thus should the pointer if not NULL be saved and restored below?

Reply:
Since the audit message format was changed to protobuf, this parameter is actually not needed anymore
Thus, i removed it completely, thanks for spotting this